fearless/Modules/Staking/Stories/View/StoriesProgressBar.swift

```diff
@@ -0,0 +1,227 @@
+import UIKit
+import SoraUI
+import SoraFoundation
+
+protocol StoriesProgressBarDataSource: class {
+    func numberOfSegments() -> Int
+    func segmentDuration() -> TimeInterval
+}
+
+protocol StoriesProgressBarDelegate: class {
+    func didFinishSegmentAnimation()
+}
+
+@IBDesignable
+extension StoriesProgressBar {
+    @IBInspectable
+    var padding: CGFloat {
+        get {
+            return stackView.spacing
+        }
+
+        set {
+            stackView.spacing = newValue
+            setNeedsLayout()
+        }
+    }
+}
+
+class StoriesProgressBar: UIView {
+    private enum ProgressBarAnimationState {
+        case stopped
+        case animating
+        case paused
+        case interrupted
+    }
+
+    private(set) var stackView: UIStackView!
+    private(set) var currentIndex: Int = 0
+    private var timer: CountdownTimerProtocol?
+    private var remainingAnimationTime: TimeInterval = 0
+    private var animationState: ProgressBarAnimationState = .stopped
+
+    private struct Constants {
+        static let height: CGFloat = 2
+    }
+
+    private var segments: [ProgressView] = []
+    private var animationDuration: TimeInterval = 5
+
+    weak var dataSource: StoriesProgressBarDataSource?
+    weak var delegate: StoriesProgressBarDelegate?
+
+    // MARK: - Overrides
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        configure()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        configure()
+    }
+
+    override func prepareForInterfaceBuilder() {
+        super.prepareForInterfaceBuilder()
+        configure()
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        stackView.frame = bounds
+        configureSegmentsLayout()
+    }
+
+    // MARK: - Private functions
+    private func configure() {
+        backgroundColor = .clear
+
+        configureTimer()
+        configureStackViewIfNeeded()
+        stackView.frame = bounds
+    }
+
+    private func configureTimer() {
+        timer = CountdownTimer(delegate: self)
+    }
+
+    private func configureStackViewIfNeeded() {
+        guard stackView == nil else { return }
+
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
```

Comment:
.horizontal is default

Reply:
I don't rely on defaults in such cases (: 
Anyway, let's remove